ACKategories-iOSTests/FlowCoordinator/FlowCoordinatorTests.swift

```diff
@@ -0,0 +1,185 @@
+        childFC.stop { exp.fulfill() }
+        wait(for: [exp], timeout: 1)
+        
+        XCTAssertEqual(fc.navigationController?.viewControllers.count, 1)
+        XCTAssertEqual(fc.childCoordinators.count, 0)
+    }
+    
+    func testFirstNavigationSubflowStopsSecondAlso() {
+        let fc = NavigationFC()
+        fc.start(in: window)
+        _ = fc.rootViewController.view
+        
+        let childFC = InnerNavigationFC()
+        fc.addChild(childFC)
+        childFC.start(with: fc.navigationController!)
+        childFC.push(UIViewController())
+        _ = childFC.rootViewController.view
+        
+        let child2FC = InnerNavigationFC()
+        childFC.addChild(child2FC)
+        child2FC.start(with: fc.navigationController!)
+        child2FC.push(UIViewController())
+        _ = child2FC.rootViewController.view
+        
+        XCTAssertEqual(fc.navigationController?.viewControllers.count, 5)
+        XCTAssertEqual(fc.childCoordinators.count, 1)
+        XCTAssertEqual(childFC.childCoordinators.count, 1)
+        
+        let exp = expectation(description: "did stop")
+        childFC.stop { exp.fulfill() }
+        wait(for: [exp], timeout: 1)
+        
+        XCTAssertEqual(fc.navigationController?.viewControllers.count, 1)
+        XCTAssertEqual(fc.childCoordinators.count, 0)
+    }
+    
+    func testDismiss() {
+        let fc = NavigationFC()
+        fc.start(in: window)
+        _ = fc.rootViewController.view
+        
+        let exp = expectation(description: "did present")
+        let childFC = PresentFC { exp.fulfill() }
+        fc.addChild(childFC)
+        childFC.start(from: fc.rootViewController)
+        _ = childFC.rootViewController.view
+        wait(for: [exp], timeout: 1)
+
+        XCTAssertNotNil(fc.rootViewController!.presentedViewController)
+        
+        let exp2 = expectation(description: "did stop")
+        childFC.stop { exp2.fulfill() }
+        wait(for: [exp2], timeout: 1)
+        
+        XCTAssertNil(fc.rootViewController.presentedViewController)
+        XCTAssertEqual(fc.childCoordinators.count, 0)
+    }
+    
+    func testDismissMultipleFlows() {
+        let fc = NavigationFC()
+        fc.start(in: window)
+        _ = fc.rootViewController.view
+
+        let exp = expectation(description: "did present")
+        let childFC = PresentFC { exp.fulfill() }
+        fc.addChild(childFC)
+        childFC.start(from: fc.rootViewController)
+        _ = childFC.rootViewController.view
+        wait(for: [exp], timeout: 1)
+
+        XCTAssertNotNil(fc.rootViewController!.presentedViewController)
+
+        let exp2 = expectation(description: "did present")
+        let child2FC = PresentFC { exp2.fulfill() }
+        childFC.addChild(child2FC)
+        child2FC.start(from: childFC.rootViewController)
+        _ = child2FC.rootViewController.view
+        wait(for: [exp2], timeout: 1)
+
+        XCTAssertNotNil(childFC.rootViewController!.presentedViewController)
```

Comment:
the force unwrap here is unnecessary, too

Reply:
👍